drivers/hmis/app/models/hmis/filter/enrollment_filter.rb

```diff
@@ -49,4 +50,35 @@ def with_project_types(scope)
   def with_search_term(scope)
     with_filter(scope, :search_term) { scope.matching_search_term(input.search_term) }
   end
+
+  def with_household_tasks(scope)
+    with_filter(scope, :household_tasks) do
+      if input.household_tasks&.include?('ANNUAL_DUE')
+        entry_anniversary = Arel.sql('make_date(extract(year from current_date)::integer, extract(month from "EntryDate")::integer, extract(day from "EntryDate")::integer)')
+        start_date = Arel.sql('make_date(extract(year from current_date)::integer, extract(month from "EntryDate")::integer, extract(day from "EntryDate")::integer) - interval \'30 days\'')
+
+        assessment_ids = Hmis::Hud::CustomAssessment.
+          where(
+            enrollment_id: scope.pluck(:enrollment_id),
+            data_source_id: scope.joins(:project).pluck(p_t[:data_source_id]).uniq,
+            data_collection_stage: 5,
+            wip: false,
+          ).
+          pluck(:id)
```

Comment:
This works but the 'pluck' pattern doesn't scale very well. We can leave this as is and come back to it but I'd suggest we avoid this in the future.